ui/src/app/highlight-search/highlight-search.component.spec.ts

```diff
@@ -5,21 +5,25 @@ import { HighlightSearchComponent } from './highlight-search.component';
 describe('HighlightSearchComponent', () => {
   let component: HighlightSearchComponent;
   let fixture: ComponentFixture<HighlightSearchComponent>;
-
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ HighlightSearchComponent ]
-    })
-    .compileComponents();
+        declarations: [ HighlightSearchComponent ]
+      })
+      .compileComponents();
   }));
-
   beforeEach(() => {
     fixture = TestBed.createComponent(HighlightSearchComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
-
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+  it('highlight_split_test', () => {
```

Comment:
Some suggestions here: https://github.com/all-of-us/workbench/compare/SrushtiGangireddy/RW-1757...ch/hilight-test?expand=1

At a high level the changes are:
- Actually render the component (use a test wrapper component to achieve this)
- Try to make assertions about the actually rendered HTML, rather then probing internal methods

I'm a bit confused by the output though. Seems like there's a lot of extra spans being produced here. I'm wondering whether you need this two-level behavior at all in the component. Could you just split the whole "searchWords" string by using the searchTerm matchString regexp?